src/katgpucbf/xbgpu/correlation.py

```diff
@@ -199,6 +199,11 @@ def __init__(
         self.slots["in_samples"] = accel.IOSlot(dimensions=input_data_dimensions, dtype=np.int8)
         self.slots["mid_visibilities"] = accel.IOSlot(dimensions=mid_data_dimensions, dtype=np.int64)
         self.slots["out_visibilities"] = accel.IOSlot(dimensions=mid_data_dimensions[1:], dtype=np.int32)
+        self.slots["out_saturated"] = accel.IOSlot(dimensions=(), dtype=np.uint32)
```

Comment:
Out of interest, what does indicating 'empty dimensions' actually do?

Reply:
It's zero-dimensional, so effectively scalar (although numpy has some subtle but occasionally annoying differences between true scalars and zero-dimensional arrays).